driver/bpf/fillers.h

```diff
@@ -2314,138 +2314,164 @@ FILLER(proc_startupdate, true)
+			CHECK_RES(res);
 		}
-	} else if (data->state->tail_ctx.evt_type == PPME_SYSCALL_EXECVE_19_X ||
-	           data->state->tail_ctx.evt_type == PPME_SYSCALL_EXECVEAT_X ) {
-		unsigned long val;
-		char **argv;
-
-		switch (data->state->tail_ctx.evt_type)
+		else /* Failure */
 		{
-		case PPME_SYSCALL_EXECVE_19_X:
-			val = bpf_syscall_get_argument(data, 1);
-			break;
+			char **argv = NULL;
+			switch(type)
+			{
+			case PPME_SYSCALL_EXECVE_19_X:
+				argv = (char **)bpf_syscall_get_argument(data, 1);
+				break;
 
-		case PPME_SYSCALL_EXECVEAT_X:
-			val = bpf_syscall_get_argument(data, 2);
-			break;
+			case PPME_SYSCALL_EXECVEAT_X:
+				argv = (char **)bpf_syscall_get_argument(data, 2);
+				break;
 
-		default:
-			val = 0;
-			break;
-		}
-		argv = (char **)val;
+			default:
+				break;
+			}
 
-		res = bpf_accumulate_argv_or_env(data, argv, &args_len);
-		if (res != PPM_SUCCESS)
-			args_len = 0;
-	} else {
-		args_len = 0;
+			unsigned long total_args_len = 0;
+			/* Here we put already all we need into our auxmap */
+			res = bpf_accumulate_argv_or_env(data, argv, &total_args_len);
+			if(res != PPM_SUCCESS)
+				total_args_len = 0;
+
+			u16 exe_arg_len = bpf_probe_read_kernel_str(&data->buf[data->state->tail_ctx.curoff & SCRATCH_SIZE_HALF],
+								    (total_args_len & SCRATCH_SIZE_HALF),
+								    &data->buf[data->state->tail_ctx.curoff & SCRATCH_SIZE_HALF]);
+
+			/* Parameter 2: exe (type: PT_CHARBUF) */
+			data->curarg_already_on_frame = true;
+			res = __bpf_val_to_ring(data, 0, exe_arg_len, PT_CHARBUF, -1, false, KERNEL);
+			CHECK_RES(res);
+
+			/* Parameter 3: args (type: PT_CHARBUFARRAY) */
+			data->curarg_already_on_frame = true;
+			res = __bpf_val_to_ring(data, 0, (total_args_len - exe_arg_len) & SCRATCH_SIZE_HALF, PT_BYTEBUF, -1, false, KERNEL);
+			CHECK_RES(res);
+		}
 	}
+	else
+	{
+		// PPME_SYSCALL_CLONE_20_X
+		// PPME_SYSCALL_FORK_20_X
+		// PPME_SYSCALL_VFORK_20_X
+		// PPME_SYSCALL_CLONE3_X
+		if(retval >= 0)
+		{
+			unsigned long int start_pointer = 0;
+			unsigned long int end_pointer = 0;
+			unsigned long int total_len = 0;
+			bpf_probe_read_kernel(&start_pointer, sizeof(start_pointer), &mm->arg_start);
+			bpf_probe_read_kernel(&end_pointer, sizeof(end_pointer), &mm->arg_end);
+			total_len = end_pointer - start_pointer;
 
-	if (args_len > 0) {
-		int exe_len;
+#ifdef BPF_FORBIDS_ZERO_ACCESS
+			bpf_probe_read_user(&data->buf[data->state->tail_ctx.curoff & SCRATCH_SIZE_HALF],
+					    ((total_len - 1) & SCRATCH_SIZE_HALF) + 1,
+					    (void *)start_pointer);
+#else
+			bpf_probe_read_user(&data->buf[data->state->tail_ctx.curoff & SCRATCH_SIZE_HALF],
+					    total_len & SCRATCH_SIZE_HALF,
+					    (void *)start_pointer);
+#endif
 
-		exe_len = bpf_probe_read_kernel_str(&data->buf[data->state->tail_ctx.curoff & SCRATCH_SIZE_HALF],
-						    SCRATCH_SIZE_HALF,
-						    &data->buf[data->state->tail_ctx.curoff & SCRATCH_SIZE_HALF]);
+			/* if true application is using setproctitle() */
```

Comment:
Just to better highlight the algo here: https://elixir.bootlin.com/linux/latest/source/mm/util.c#L965